rotkehlchen/chain/ethereum/modules/balancer/balancer.py

```diff
@@ -157,12 +157,14 @@ def _calculate_pool_events_balances(
                 pool_addr_to_usd_value[pool_balance.address] += pool_balance.tokens[idx].user_balance.usd_value  # noqa: E501
 
         for pool_address, pool_events in pool_addr_to_events.items():
+            pool_tokens = pool_addr_to_pool[pool_address].tokens
+            profit_loss_amounts = pool_addr_to_profit_loss_amounts[pool_address][:len(pool_tokens)]
```

Comment:
`profit_loss_amounts` had a list of 8 zero amounts. For consistency the length of this list must be the number of tokens in the pool.

Reply:
Because of the initialization you are making here:

```python
pool_addr_to_profit_loss_amounts: DDAddressToProfitLossAmounts = (
            defaultdict(lambda: [AssetAmount(ZERO)] * POOL_MAX_NUMBER_TOKENS)
        )
```

?

Reply:
Yes. It wasn't like this when I wasn't using the lambda on the defaultdict. But I introduced the lambda as I wanted to reduce the code.